include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -1,13 +1,45 @@
 #ifndef _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
 #define _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
 
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/fft/common.hlsl>
+
+#ifndef __HLSL_VERSION
+#include <nbl/video/IPhysicalDevice.h>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace workgroup
+{
+namespace fft
+{
+
+inline std::pair<uint16_t, uint16_t> optimalFFTParameters(const video::ILogicalDevice* device, uint32_t inputArrayLength)
+{
+    uint32_t maxWorkgroupSize = *device->getPhysicalDevice()->getLimits().maxWorkgroupSize;
+    // This is the logic found in core::roundUpToPoT to get the log2
+    uint16_t workgroupSizeLog2 = 1u + hlsl::findMSB(core::min(inputArrayLength / 2, maxWorkgroupSize) - 1u);
+    uint16_t elementPerInvocationLog2 = 1u + hlsl::findMSB(core::max((inputArrayLength >> workgroupSizeLog2) - 1u, 1u));
+    return { elementPerInvocationLog2, workgroupSizeLog2 };
+}
+
+}
+}
+}
+}
+
+#else
```

Comment:
HLSL has no findMSB, I implemented the SPIR-V intrinsics in the std.450 instruction set and the GLSL equivalents in `glsl_compat/core.hlsl` but not sure how to unify them since one in in `hlsl` namespace and the other is in `hlsl::glsl`

Reply:
> HLSL has no findMSB, I implemented the SPIR-V intrinsics in the std.450 instruction set and the GLSL equivalents in `glsl_compat/core.hlsl` but not sure how to unify them since one in in `hlsl` namespace and the other is in `hlsl::glsl`

@Przemog1 will fix in #801